### PR TITLE
Migrating Store Grid to UI Components

### LINKED ISSUE #7283: Insignificant stores sorting issue
### Preconditions
1. Magento 2.1.2 Release Archive 
2. Magento 2.4-develop

### Steps to reproduce
1. Login to Admin
2. go to Stores -> All Stores.
3. Sort by any column

### Expected result
Sorting should work or disable the  sorting option

### Actual result
sorting doesn't work

### PR DESCRIPTION
**Pull request by afirlejczyk**
### Description

Fix sorting by columns in store listing.

I override function `joinGroupAndStore` in `\Magento\Store\Model\ResourceModel\Website\Grid\Collection`.
I also use UI component to render store listing instead of block.

### Fixed Issues (if relevant)

1. magento/magento2#7283: Insignificant stores sorting issue

### Manual testing scenarios

1. Login to Admin.
2. Go to Stores -> All Stores.
3. Sort by any column.
4. Sorting should work.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=17371) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @afirlejczyk. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation

**Comment by afirlejczyk** [comment]
I see some tests failed, I will fix this.

File: app/code/Magento/Backend/Block/System/Store/Grid/Render/Group.php [hunk 1]
```diff
@@ -9,6 +9,7 @@
  * Store render group
  *
  * @author      Magento Core Team <core@magentocommerce.com>
+ * @deprecated
```
Comment (slavvka) [inline app/code/Magento/Backend/Block/System/Store/Grid/Render/Group.php]: Please mention the reason of deprecation of the class. See here https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/index.html#deprecation

File: app/code/Magento/Backend/Block/System/Store/Grid/Render/Store.php [hunk 2]
```diff
@@ -9,6 +9,7 @@
  * Store render store
  *
  * @author      Magento Core Team <core@magentocommerce.com>
+ * @deprecated
```
Comment (slavvka) [inline app/code/Magento/Backend/Block/System/Store/Grid/Render/Store.php]: Please mention the reason of deprecation of the class. See here https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/index.html#deprecation

File: app/code/Magento/Backend/Block/System/Store/Grid/Render/Website.php [hunk 3]
```diff
@@ -9,6 +9,7 @@
  * Store render website
  *
  * @author      Magento Core Team <core@magentocommerce.com>
+ * @deprecated
```
Comment (slavvka) [inline app/code/Magento/Backend/Block/System/Store/Grid/Render/Website.php]: Please mention the reason of deprecation of the class. See here https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/index.html#deprecation

File: app/code/Magento/Backend/Block/System/Store/Store.php [hunk 4]
```diff
@@ -12,6 +12,7 @@
  * @author      Magento Core Team <core@magentocommerce.com>
  * @api
  * @since 100.0.2
+ * @deprecated
```
Comment (slavvka) [inline app/code/Magento/Backend/Block/System/Store/Store.php]: Please mention the reason of deprecation of the class. See here https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/index.html#deprecation

File: app/code/Magento/Store/Ui/Component/Listing/Column/AbstractNameColumn.php [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Store\Ui\Component\Listing\Column;
+
+/**
+ * Class AbstractNameColumn
+ */
+abstract class AbstractNameColumn extends \Magento\Ui\Component\Listing\Columns\Column
```
Comment (slavvka) [inline app/code/Magento/Store/Ui/Component/Listing/Column/AbstractNameColumn.php]: I believe you reused the code from other Magento module. But not all the legacy code currently fits Magento Technical Guideline (https://devdocs.magento.com/guides/v2.2/coding-standards/technical-guidelines.html#2-class-design). Please try to reimplement this using Composition if possible

**Review by slavvka — CHANGES_REQUESTED** [review]
Please fix the issues

**Comment by slavvka** [comment]
Hey @afirlejczyk Thank you for this huge and meaningful work! But currently there are conflicts with the latest magento code so please merge it and resolve conflicts. Also please look at the changes requested and fix them.

**Review by slavvka — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @slavvka, thank you for the review.
[ENGCOM-2667](https://jira.corp.magento.com/browse/ENGCOM-2667) has been created to process this Pull Request

**Comment by afirlejczyk** [comment]
Hi @slavvka, thank you for adding changes!

**Comment by slavvka** [comment]
@afirlejczyk Sure :)

**Comment by slavvka** [comment]
Hey @afirlejczyk Our internal tests found inconsistency in UI definition:

```Magento\Test\Integrity\UiComponentAclTest::testThatAclExistsInUiComponentDataSource
Passed: 26, Failed: 1, Incomplete: 0, Skipped: 0.
Data set: array (
  'file' => '/var/www/html/app/code/Magento/Store/view/adminhtml/ui_component/store_listing.xml',
  'dataSourceXml' => 
  SimpleXMLElement::__set_state(array(
     '@attributes' => 
    array (
      'name' => 'store_listing_data_source',
      'component' => 'Magento_Ui/js/grid/provider',
    ),
     'settings' => 
    SimpleXMLElement::__set_state(array(
       'storageConfig' => 
      SimpleXMLElement::__set_state(array(
         'param' => 'store_id',
      )),
       'updateUrl' => 
      SimpleXMLElement::__set_state(array(
         '@attributes' => 
        array (
          'path' => 'mui/index/render',
        ),
      )),
    )),
     'dataProvider' => 
    SimpleXMLElement::__set_state(array(
       '@attributes' => 
      array (
        'class' => 'Magento\\Store\\Ui\\DataProvider\\WebsiteDataProvider',
        'name' => 'store_listing_data_source',
      ),
       'settings' => 
      SimpleXMLElement::__set_state(array(
         'requestFieldName' => 'id',
         'primaryFieldName' => 'website_id',
      )),
    )),
  )),
)
UI Component '/var/www/html/app/code/Magento/Store/view/adminhtml/ui_component/store_listing.xml' contains dataSource element without an aclResource```


Could you please fix that?

File: app/code/Magento/Store/Ui/DataProvider/WebsiteDataProvider.php [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+/**
+ * Class WebsiteDataProvider
+ */
+class WebsiteDataProvider extends \Magento\Ui\DataProvider\AbstractDataProvider
+{
+    /**
+     * @var WebsiteCollection
+     */
+    protected $collection;
```
Comment (slavvka) [inline app/code/Magento/Store/Ui/DataProvider/WebsiteDataProvider.php]: Please change the type to `private`

**Review by slavvka — CHANGES_REQUESTED** [review]

**Comment by slavvka** [comment]
Hey @afirlejczyk are you going to proceed with this PR?

File: app/code/Magento/Store/Ui/DataProvider/WebsiteDataProvider.php (hunk 6 above)
Comment (afirlejczyk) [reply app/code/Magento/Store/Ui/DataProvider/WebsiteDataProvider.php]: It is unnecessary here, I will remove this. 
`$collection` is declared in parent class.

**Comment by afirlejczyk** [comment]
@slavvka yes, I will try to make changes tomorrow or day after tomorrow.

**Comment by slavvka** [comment]
Hey @afirlejczyk Are you going to proceed with the PR? It is very valuable change that would be very good to complete since there are only few changes left.

File: app/code/Magento/Store/Ui/DataProvider/WebsiteDataProvider.php (hunk 6 above)
Comment (p-bystritsky) [reply app/code/Magento/Store/Ui/DataProvider/WebsiteDataProvider.php]: @slavvka removed.

**Comment by p-bystritsky** [comment]
> 
> 
> Hey @afirlejczyk Our internal tests found inconsistency in UI definition:
> 
> ```
> Passed: 26, Failed: 1, Incomplete: 0, Skipped: 0.
> Data set: array (
>   'file' => '/var/www/html/app/code/Magento/Store/view/adminhtml/ui_component/store_listing.xml',
>   'dataSourceXml' => 
>   SimpleXMLElement::__set_state(array(
>      '@attributes' => 
>     array (
>       'name' => 'store_listing_data_source',
>       'component' => 'Magento_Ui/js/grid/provider',
>     ),
>      'settings' => 
>     SimpleXMLElement::__set_state(array(
>        'storageConfig' => 
>       SimpleXMLElement::__set_state(array(
>          'param' => 'store_id',
>       )),
>        'updateUrl' => 
>       SimpleXMLElement::__set_state(array(
>          '@attributes' => 
>         array (
>           'path' => 'mui/index/render',
>         ),
>       )),
>     )),
>      'dataProvider' => 
>     SimpleXMLElement::__set_state(array(
>        '@attributes' => 
>       array (
>         'class' => 'Magento\\Store\\Ui\\DataProvider\\WebsiteDataProvider',
>         'name' => 'store_listing_data_source',
>       ),
>        'settings' => 
>       SimpleXMLElement::__set_state(array(
>          'requestFieldName' => 'id',
>          'primaryFieldName' => 'website_id',
>       )),
>     )),
>   )),
> )
> UI Component '/var/www/html/app/code/Magento/Store/view/adminhtml/ui_component/store_listing.xml' contains dataSource element without an aclResource```
> 
> 
> Could you please fix that?
> ```

Done.

**Pull request merged**